components/author/index.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { useContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import PropTypes from 'prop-types';
+import { AUTHOR_CONTEXT } from './context';
+
+/**
+ * @typedef {object} Author
+ * @property {object} author
+ * @property {object<string, string>} author.avatar_urls
+ * @property {string} author.description
+ * @property {string} author.email
+ * @property {string} author.first_name
+ * @property {number} author.id
+ * @property {string} author.last_name
+ * @property {string} author.link
+ * @property {string} author.name
+ * @property {string} author.nickname
+ * @property {string} author.registered_date
+ * @property {string} author.slug
+ * @property {string} author.url
+ */
+
+export const Name = (props) => {
+	const { tagName: TagName = 'span', ...rest } = props;
+
+	/**
+	 * @type {Author}
+	 */
+	const { name, link } = useContext(AUTHOR_CONTEXT);
+
+	const wrapperProps = { ...rest };
+
+	if (TagName === 'a' && link) {
+		wrapperProps.href = link;
+	}
+
+	return <TagName {...wrapperProps}>{name}</TagName>;
+};
+
+Name.propTypes = {
+	tagName: PropTypes.string,
+};
+
+Name.defaultProps = {
+	tagName: 'span',
+};
+
+export const FirstName = (props) => {
+	const { tagName: TagName = 'span', ...rest } = props;
```

Comment:
[nit] Don't need a default here since it's added below on `defaultProps`

Reply:
Fixed in [35d0436](https://github.com/10up/block-components/pull/178/commits/35d043649878155c360b3a65f92d6271f2c3174e)